twisted/test/proto_helpers.py

```diff
@@ -142,6 +142,13 @@ class StringTransport:
     @ivar io: A L{io.BytesIO} which holds the data which has been written to
         this transport since the last call to L{clear}.  Use L{value} instead
         of accessing this directly.
+
+    @ivar lenient: A L{bool} that controls whether L{StringTransport} will
```

Comment:
For what it's worth: I'm OK with the flag but I'm more or less neutral on having multiple different classes.  -1 for using inheritance though ;)